### PR TITLE
Move document logic to DocumentProvider

### PR DESCRIPTION
**Pull request by yufeih**
###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5279)

### CODE REVIEW

File: src/docfx/build/legacy/LegacyUtility.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ public static string ToLegacyOutputPathRelativeToSiteBasePath(this Document doc,
             var outputPath = manifestItem.Path;
             if (doc.ContentType == ContentType.Resource && !doc.Docset.Config.Output.CopyResources)
             {
-                outputPath = doc.GetOutputPath(manifestItem.Monikers, isPage: false);
+                outputPath = UrlUtility.Combine(docset.SiteBasePath, MonikerUtility.GetGroup(manifestItem.Monikers) ?? "", doc.SitePath);
```
Comment (live1206) [inline src/docfx/build/legacy/LegacyUtility.cs]: not reuse `documentProvider.GetOutputPath` here, do not want to involve `context` as dependency?

Comment (yufeih) [reply src/docfx/build/legacy/LegacyUtility.cs]: yep

File: src/docfx/build/legacy/LegacyFileMap.cs [hunk 2]
```diff
@@ -27,7 +27,7 @@ internal static class LegacyFileMap
                     fileManifest =>
                     {
                         var document = fileManifest.Key;
-                        if (!document.IsPage)
+                        if (document.ContentType == ContentType.Page && !document.IsPage)
```
[the remark this answers is not in the document]
Comment (yufeih) [reply src/docfx/build/legacy/LegacyFileMap.cs]: `Redirection` __was__ a page, thus this logic is the same as before, may be should introduce a `ContentType`: `Data` to simplify the mess

[the remark this answers is not in the document]
Comment (live1206) [reply src/docfx/build/legacy/LegacyFileMap.cs]: add a TODO here?

**Pull request merged**